website/integrations: proxmox: cleanup guide

dominic-r opened the pull request:
Cleans up the setup guide

<!--
👋 Hi there! Welcome.

Please check the Contributing guidelines: https://docs.goauthentik.io/docs/developer-docs/#how-can-i-contribute
-->

## Details

<!--
Explain what this PR changes, what the rationale behind the change is, if any new requirements are introduced or any breaking changes caused by this PR.

Ideally also link an Issue for context that this PR will close using `closes #`
-->
REPLACE ME

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [x] The documentation has been updated
-   [x] The documentation has been formatted (`make website`)

Commit:
website/integrations: proxmox: clarify location of Providers tab

Updates the guide to explicitly state the path to the Providers tab.

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
website/integrations: proxmox: cleanup instructions

Cleans up instructions to make them more readable


Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
website/integrations: proxmox: specify application tab location

Specifies the location of the Applications tab in the admin interface

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
website/integrations: proxmox remove duplicate instructions

Some instructions were copied over in my previous commit and were never removed. This commit removes redundant instructions

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
website/integrations: proxmox: try to adhere to style guide

Touch up a few things to try to make it match

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Merge branch 'goauthentik:main' into patch-1

Commit:
website/integrations: proxmox: lint

Lint with prettier

Commit:
Merge branch 'goauthentik:main' into patch-1

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
+## authentik setup
 
-Under _Providers_, create an OAuth2/OpenID provider with these settings:
+1. **Navigate to the Admin interface, then click Providers to create an OAuth2/OpenID provider with these settings:**
```
```suggestion
1. Navigate to the Admin interface, then click **Providers** to create an OAuth2/OpenID provider with these settings:
```

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
+    - Set the **Launch URL** to `https://proxmox.company:8006`.
 
-Create an application which uses this provider. Optionally apply access restrictions to the application.
+## Proxmox VE setup (web interface)
```
```suggestion
## Proxmox VE (web interface) configuration
```

tanberry replied:
Is "(web interface)" describing the "Proxmox VE"? Hopefully I didn't mess things up by reordering...?

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
 -   `authentik.company` is the FQDN of the authentik install.
 
-### Step 1
+## authentik setup
```
```suggestion
## authentik configuration
```

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
+    - **Signing Key:** Select any available key
 
-### Step 2
+2. **Create an application using this provider**
```
```suggestion
2. Create an application using the provider.
```

tanberry replied:
I see the aesthetic/readability appeal of bolding the whole sentence, but... we don't do that in any other docs...

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
+    - Access the Proxmox VE web interface and log in with an administrative account.
 
-Proxmox VE allows configuration of authentication sources using the web interface (under Datacenter -> Permissions -> Realms).
+2. **Navigate to authentication source settings**
```
```suggestion
2. Navigate to authentication source settings.
```

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
+    - Click **Add** and select **Realm** to open the Add Realm dialog.
 
-Another way is to use the CLI. SSH into any Proxmox cluster node, and issue the following command:
+3. **Fill out the OpenID Connect settings**
```
```suggestion
3. Fill out the OpenID Connect settings.
```

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
 
-`pveum realm add authentik --type openid --issuer-url https://authentik.company/application/o/proxmox/ --client-id xxx --client-key xxx --username-claim username --autocreate 1`
+    - In the dialog that appears, fill in the following details:
+        - **Issuer URL**: Enter the Issuer URL from authentik (found under _Provider Metadata_), e.g., `https://authentik.company/application/o/proxmox/`.
```
```suggestion
        - **Issuer URL**: Enter the Issuer URL from authentik (found under **Provider Metadata**), e.g., `https://authentik.company/application/o/proxmox/`.
```

tanberry replied:
Where exactly is Issuer Metadata? Is that a field name somewhere on authentik's UI? Let's describe where to find this value a bit better.

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
+        - **Issuer URL**: Enter the Issuer URL from authentik (found under _Provider Metadata_), e.g., `https://authentik.company/application/o/proxmox/`.
+        - **Realm**: Enter a name for this authentication source, such as `authentik`.
+        - **Client ID**: Enter the Client ID found on the provider overview page.
+        - **Client Key**: Enter the Client Secret found by clicking _Edit_ on the Provider overview page.
```
```suggestion
        - **Client Key**: Enter the Client Secret found by clicking **Edit** on the Provider overview page.
```

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,63 @@ The following placeholders will be used:
+
+    ![Proxmox login page with authentik marked as default login method](proxmox-login.png)
+
+## Proxmox VE setup (CLI)
```
```suggestion
## Proxmox VE configuration (using CLI)
```

tanberry reviewed (changes requested):
Thanks so much @4d62 for this contribution. A couple of edits and a question... thanks again!

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Merge branch 'goauthentik:main' into patch-1

Commit:
website/integrations: proxmox: address review comment

Address review comment https://github.com/goauthentik/authentik/pull/11894#discussion_r1837106698 - Merge lines 42 and 44

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
website/integrations: proxmox: lint

run `npx prettier --write website/integrations/services/proxmox-ve/index.md`

Commit:
Merge branch 'goauthentik:main' into patch-1

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,61 @@ The following placeholders will be used:
+## authentik configuration
 
-Under _Providers_, create an OAuth2/OpenID provider with these settings:
+1. Navigate to the Admin interface, then click **Providers** to create an OAuth2/OpenID provider with these settings:
```
```suggestion
1. In the Admin interface, navigate to **Applications -> Providers** to create an OAuth2/OpenID provider with these settings:
```

tanberry commented on website/integrations/services/proxmox-ve/index.md:
```diff
@@ -24,36 +24,61 @@ The following placeholders will be used:
-After configuring the source in Proxmox, any user that logs in to Proxmox for the first time automatically gets an user named `<authentik username>@<pve realm name>`. In this example,
-authentik user `bob` will get an user named `bob@authentik` in Proxmox. You can then assign Permissions as normally in Proxmox. You can also pre-create the users in Proxmox if you want
-the user to be able to perform actions immediately after first login.
+4. **Save the configuration**
```
```suggestion
4. **Save the configuration**.
```

tanberry reviewed (changes requested):
Two tiny edits, then let's get this merged!

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Update website/integrations/services/proxmox-ve/index.md

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Merge branch 'main' into patch-1

Signed-off-by: 4d62 <github-user@sdko.org>

Commit:
Merge branch 'goauthentik:main' into patch-1

Commit:
Merge branch 'goauthentik:main' into patch-1

Commit:
Merge branch 'goauthentik:main' into patch-1

Commit:
Merge branch 'goauthentik:main' into patch-1

tanberry commented:
Thank you so much @4d62 both for this contribution and for your patience. ;-)
